src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java

```diff
@@ -42,11 +43,8 @@
     /** The tokens the check is interested in. */
     private final Set<String> tokens = new HashSet<>();
 
-    /** The sorted set for collecting messages. */
-    private final SortedSet<LocalizedMessage> messages = new TreeSet<>();
-
-    /** The current file contents. */
-    private FileContents fileContents;
+    /** The check context. */
+    private final CheckContextHolder context = new CheckContextHolder();
```

Comment:
Also since it contains contents of a file, file violations and file contents, shouldn't it be named something like `FileContextHolder`?
Also what about `AbstractFileSets`, do they require a `ThreadLocal` like this?